### DISCUSSION TITLE
[Help] Activity tracking has completely tanked - V7 Ghosts

### DISCUSSION THREAD

**Discussion by tm6907** [cmu-sei/GHOSTS#320] [Q&A] [answered]
Hi, opening another discussion sorry. Just really struggling to get it working consistently. 

Activity tracking on my clients has completely died. Got three clients running away but nothing is recorded on the Grafana dash despite all three being checked in and polling. I've tried varied time ranges. It was working at one point but inconsistently - now it has stopped all together. 

Nothing indicating any errors in the logs other than the repeating line "Quartz.Logging... Batch Acquisition of 0 triggers"

<img width="1314" alt="image" src="https://github.com/cmu-sei/GHOSTS/assets/162140616/b61df5c9-1e69-4c23-9941-c001c61d157e">

**Comment by sei-dupdyke** [comment]
Go to your server machine:

```
docker exec -it ghosts-postgres /bin/bash

psql -U ghosts

select id, machineid, createdutc, now() from history_timeline order by id desc limit 5;
```

Do you see records? What is the time of last check in?

**Reply by tm6907** [reply]
Yeah get nothing returned 
<img width="809" alt="image" src="https://github.com/cmu-sei/GHOSTS/assets/162140616/9ed5838c-fd38-4a12-b1d6-7d1d91e65cd0">
There should be at least some firefox and ms office activity from 2 clients.

They're definitely checked in and polling but nothing else. 
<img width="1310" alt="image" src="https://github.com/cmu-sei/GHOSTS/assets/162140616/bffcd6e0-9d47-49e1-b03e-2b11a507d058">
<img width="1310" alt="image" src="https://github.com/cmu-sei/GHOSTS/assets/162140616/eb2c35aa-aab7-4239-ac96-59b9e8826b4b">

**Reply by sei-dupdyke** [reply]
Hmm, let's try another couple of queries:

```
docker exec -it ghosts-postgres /bin/bash

psql -U ghosts

select id, machineid, createdutc, now() from history_timeline order by id desc limit 5;
select id, machineid, createdutc, now() from history_health order by id desc limit 5;
select id, machineid, createdutc, now() from history_machine order by id desc limit 5;
```

If still zeros all around, let's look at docker logs:

```
docker logs --tail 100 -f ghosts-api 
```

and watch it for a bit... any errors or failures?

**Reply by tm6907** [reply]
Only the 3rd query had results. I left the clients running overnight and no change - I did have to stop and restart my rocky8 client because it filled the /tmp folder and had really high CPU usage. 

<img width="1310" alt="image" src="https://github.com/cmu-sei/GHOSTS/assets/162140616/ea9cc281-f414-4035-958b-79068e9b3348">

Seems my win7 client also isn't working so just a rocky8 and win10 client should be showing activity. Not sure why health history is empty because health check is enabled on each client and has worked previously. Seems like something just broke, not sure what :/

Checking docker logs just yields the same repeated message 
<img width="1310" alt="image" src="https://github.com/cmu-sei/GHOSTS/assets/162140616/6dcf2e59-1533-4634-aae1-66eec06f5123">

Might be something on Grafana, It sometimes shows errors at one point I had to migrate to a later version of something. This is the sql.
<img width="1310" alt="image" src="https://github.com/cmu-sei/GHOSTS/assets/162140616/c0b50c1a-809a-492e-a5f7-5deae74f3939">

**Reply by sei-dupdyke** [reply, marked as answer]
Ugh, sorry!

I am not sure what is going on here. Can I potentially convince you to give v8 a try?

**Reply by tm6907** [reply]
Yeah gonna give it a try now, will come back if I run into any issues

**Reply by capstonewazuhelk** [reply]
Hi, I have the same issue @sei-dupdyke ,

docker logs --tail 100 -f ghosts-api 
![image](https://github.com/cmu-sei/GHOSTS/assets/147314420/24eac6fd-34bd-4f7d-a570-e4fe19ab886a)

why there is no data history_timelines in the postgresql and activity in grafana ? help me please
